### ISSUE TITLE
Server crashed on joining?

### ISSUE THREAD

**Issue by ideaman924** [mroote/factorio-server-manager#165]
Hi all, the server just crashed and this came out on the terminal:

```
2019/10/30 15:59:17 Factorio Server: 21525.002 ConnectionRequestReplyConfirm (type(ConnectionRequestReplyConfirm) ) from(IP ADDR:({104.247.132.212:49725}))
panic: send on closed channel

goroutine 100 [running]:
main.logSubscribe.func1(0xc00001f200)
        /home/mitch/src/go/src/github.com/mroote/factorio-server-manager/src/wsroutes.go:19 +0x17e
created by main.logSubscribe
        /home/mitch/src/go/src/github.com/mroote/factorio-server-manager/src/wsroutes.go:10 +0x3f
```

At the time the server had been in a pause state for about 5 hours (since nobody was logged on). When I tried to log in (not via WebUI, via Factorio) the game appeared to be loading (showed me the progress bar of Connecting to the server) but then said it couldn't establish a connection with the server. The above was written to the console, and then the process exited.

I ran the executable again and the server is running fine now, but any idea what this error is?

**Comment by knoxfighter** [comment]
For me, that seems like a factorio error. The factorio-server-manager tries to call something on the Server, but the server isn't running anymore. Above this error, should be the factorio output with the error.
If you dont find it, you can send the complete log here and i try to find your error.

**Comment by knoxfighter** [comment]
I never looked at that websocket code :(
It seems, like the server had problems, sending a new line in the factorio.log to the webclient.
If that happens often, i will look into that sooner. I wanted to rework the logging in the webclient nonetheless, but didnt had time yet.

**Comment by ideaman924** [comment]
Second crash, happened a couple minutes ago.

```
2019/10/31 03:22:45 listing users: 1 found
2019/10/31 03:22:58 Factorio Server: 40966.299 Info ServerMultiplayerManager.cpp:1039: Disconnect notification for peer (5)
2019/10/31 03:22:58 Factorio Server: 40966.300 Info ServerSynchronizer.cpp:618: nextHeartbeatSequenceNumber(1213232) removing peer(5).
2019/10/31 05:32:26 Factorio Server: 48734.860 ConnectionRequestReplyConfirm (type(ConnectionRequestReplyConfirm) ) from(IP ADDR:({59.46.193.106:12133}))
panic: send on closed channel

goroutine 144 [running]:
main.logSubscribe.func1(0xc00054b620)
        /home/mitch/src/go/src/github.com/mroote/factorio-server-manager/src/wsroutes.go:19 +0x17e
created by main.logSubscribe
        /home/mitch/src/go/src/github.com/mroote/factorio-server-manager/src/wsroutes.go:10 +0x3f
ideaman924@factorio:~/factorio-server-manager$
```

Any idea @knoxfighter ?

**Comment by ideaman924** [comment]
The issue seems to be triggered if the player is on a bad Internet connection. Something happens in the communication process between the Factorio server and the player, and it tries to log it but fails because of something.

**Comment by mroote** [comment]
It looks like the user left Factorio but the websocket process is still trying to send to a channel that's now closed. So we will need to close that channel after a player disconnects from a Factorio server.

**Comment by ideaman924** [comment]
Is there a way to run the server while bypassing the manager? It crashes every time someone has a bad Internet connection. Preferably keeping the configuration settings in the manager so that it autosaves every 5 min and what not.

**Comment by knoxfighter** [comment]
You can just start the factorio server directly. https://wiki.factorio.com/Multiplayer
Simple start: `./bin/x64/factorio --start-server ./saves/my-save.zip`

@mroote I dont really understand that code yet. And ever worse, i dont understand how that code is even relevant, when somebody is joining the server.

**Comment by knoxfighter** [comment]
You noticed that yourself, you can define the settingsfile yourself. We use the default paths, so it is not really necessary. And yes, that should work, until we have that bug fixed.

**Comment by ideaman924** [comment]
Did not know about that! So in theory, this should be the only thing people need to run the Factorio server manually?

`./bin/x64/factorio --start-server-load-latest`

Sounds excellent. Hope the bug gets fixed soon. Thanks for the great manager interface, really loved using it so far!

**Comment by knoxfighter** [comment]
This has been fixed with #178
